cmd/dockerd/daemon.go

```diff
@@ -151,6 +158,12 @@ func (cli *DaemonCli) start(opts daemonOptions) (err error) {
 		}
 	}
 
+	// Create the daemon root before we create ANY other files (PID, or migrate keys)
+	// to ensure the appropriate ACL is set (particularly relevant on Windows)
+	if err := daemon.CreateDaemonRoot(cli.Config); err != nil {
+		return err
+	}
+
```

Comment:
Same question: can we keep this inside `NewDaemon`. From go api standpoint it doesn't make sense that this preparation is needed before calling `NewDaemon`.